src/dbnode/storage/shard.go

```diff
@@ -342,7 +337,7 @@ func (s *dbShard) Stream(
 // IsBlockRetrievable implements series.QueryableBlockRetriever
 func (s *dbShard) IsBlockRetrievable(blockStart time.Time) bool {
 	flushState := s.FlushState(blockStart)
-	return statusIsRetrievable(flushState.Status)
+	return statusIsRetrievable(flushState.WarmStatus) || flushState.ColdVersion > 0
```

Comment:
This would only happen in the case of future writes.

Reply:
I still don't think this logic works for `FutureWrites` because if the node restarted after performing a cold future write, the ColdVersion won't be restored after restarting the node and bootstrapping will it? Maybe file an issue to track